pkg/ddevapp/craftcms.go

```diff
@@ -0,0 +1,170 @@
+package ddevapp
+
+import (
+	"fmt"
+	"github.com/drud/ddev/pkg/archive"
+	"github.com/drud/ddev/pkg/fileutil"
+	"github.com/drud/ddev/pkg/util"
+	"github.com/pkg/errors"
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+// isCraftCmsApp returns true if the app is of type craftcms
+func isCraftCmsApp(app *DdevApp) bool {
+	return fileutil.FileExists(filepath.Join(app.AppRoot, "craft"))
+}
+
+// Set the Docroot to web
+func craftCmsConfigOverrideAction(app *DdevApp) error {
+	if app.Docroot == "" {
+		app.Docroot = "web"
+	}
+
+	return nil
+}
+
+// Returns the upload directory for importing files, if not already set
+func getCraftCmsUploadDir(app *DdevApp) string {
+	app.UploadDir = "files"
```

Comment:
I assume this needs some comments explaining Craft behavior here because this really isn't the default is it? I thought there was no default?

Reply:
There is no default, but if we don't provide this, tests do not work for `import-files` -- so it seemed as good as anything. I understand you have a potential FR in the works that may improve on DDEV's assumption of a singleton upload directory, which would help us too